### PR TITLE
Playwright POM

### PR DESCRIPTION
**Pull request by BOHEUS**
Related to #6641

### CODE REVIEW

File: packages/twenty-e2e-testing/tests/login.setup.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { test as setup, expect } from '@playwright/test';
-import path from 'path';
+import { config } from 'dotenv';
+import path = require('path');
+config({ path: path.resolve(__dirname, '..', '.env') });
```
Comment (Devessier) [inline packages/twenty-e2e-testing/tests/login.setup.ts:4]: I'm unsure if we want to rely on an environment variable in an E2E test, and I think we don't want them to be parameterizable.

I would create "mock" data for e2e tests like this:

```ts
const CURRENT_USER = {
  email: 'xx@xx.com',
  password: '1234',
}
```

Relying on environment variables is unsafe. What happens if the `DEFAULT_LOGIN` is set as an empty string or is not defined?

**Review by Devessier — COMMENTED** [review]
Thank you a lot for your PR, @BOHEUS. I wrote a few comments.

As discussed previously, using XPath or other "bad practices" regarding Playwright's conventions is not a problem. The goal is to make e2e tests work, and we should unlock issues.

I wonder why we start by writing abstractions instead of focusing on writing e2e tests. Did you write these abstractions alongside some specific e2e tests? I prefer that we write e2e tests with duplicate instructions instead of going all-in abstractions. Abstractions should be constructed after a few iterations and after identifying them as necessary.

I'm not saying your abstractions are wrong. We should focus on writing e2e tests. I'm okay with you writing e2e tests and creating a POM for them in the same PR. However, we prefer to avoid creating PRs to prepare for future work. We like to work in pragmatic iterations, adding some working e2e tests each time. This will make it much easier for us to review and for you to work on the subject. Don't overwhelm yourself too much.

The problem with premature abstractions is they might need to be more robust for future use. Maybe some of the selectors you wrote won't work anymore after a few actions on the page. How would we do then? Would we create `this.workspaceDropdown` and `this.workspaceDropdown2`, with different locators? I prefer we duplicate things until we see a clear and robust abstraction. We can take time for that, and [avoid hasty abstractions](https://kentcdodds.com/blog/aha-programming).

To summarize my thoughts, please start small and not add too many abstractions.

File: packages/twenty-e2e-testing/lib/pom/settings/generalSection.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import { Locator, Page } from '@playwright/test';
+
+export class generalSection {
```
[the remark this answers is not in the document]
Comment (BOHEUS) [reply packages/twenty-e2e-testing/lib/pom/settings/generalSection.ts:3]: Thanks! Totally overlooked it :eyes:

File: packages/twenty-e2e-testing/tests/login.setup.ts (hunk 1 above)
Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/login.setup.ts:4]: If `DEFAULT_LOGIN` is not defined or empty, then Playwright will throw an error `Error: locator.fill: value: expected string, got undefined at .../login.setup.ts:9:40`

Comment (BOHEUS) [reply packages/twenty-e2e-testing/tests/login.setup.ts]: I deleted this, the reason why I added these lines was thanks to wrong setup on my side - working directory changed to root folder of project where it should be a root folder of E2E testing package

**Comment by charlesBochet** [comment]
**Note:** We are a bit late on reviews! I'm attributing the oss-gg points to all opened PRs as Hacktoberfest event ends tonight. We will review the PR itself shortly

**Comment by lucasbordeau** [comment]
@BOHEUS @Devessier I'm merging this as is because we need to move forward on this big topic. We'll simplify this if necessary and if we feel like it's too much too maintain on the long run.

**Review by lucasbordeau — APPROVED** [review]
